### PR TITLE
Dynamic, CGLib and Javassist proxy objects as sources

### PR DESCRIPTION
**Pull request by naharoo**
This PR contains changes to support Dynamic, CGLib and Javassist proxy objects as sources for the MappingFacade.
Possible successor of https://github.com/naharoo/mapstruct-facade-spring-boot-starter/pull/4

### CODE REVIEW

File: src/main/java/com/naharoo/commons/mapstruct/MappingsRegistry.java [hunk 1]
```diff
@@ -86,4 +87,47 @@ static void clear() {
...
+        if (sourceSuperclass != null) {
+            final UnaryOperator<Object> superclassMappingFunction = retrieveInternal(
+                MappingIdentifier.from(
+                    sourceSuperclass,
+                    identifier.getDestination()
+                )
+            );
+            if (superclassMappingFunction != null) {
+                return superclassMappingFunction;
+            }
```
Comment (tcolombo08) [inline src/main/java/com/naharoo/commons/mapstruct/MappingsRegistry.java]: You could move this code to a method and reuse it in the following for block

**Comment by tcolombo08** [comment]
Hi Narek,
Thank you so much for your fast turnaround! 
This is better than my fast fix. I tested it with Java 17 and SpringBoot 2.6.6 and it works like a charm.

**Review by tcolombo08 — APPROVED** [review]
LGTM

File: src/main/java/com/naharoo/commons/mapstruct/MappingsRegistry.java (hunk 1 above)
Comment (naharoo) [reply src/main/java/com/naharoo/commons/mapstruct/MappingsRegistry.java]: I've extracted this part 👍

**Comment by naharoo** [comment]
I'm merging this. Thank you @tcolombo08 for your help 👍 
Will release this ASAP and ping you in https://github.com/naharoo/mapstruct-facade-spring-boot-starter/pull/4 thread

**Pull request merged**